### PR TITLE
[Hotwallet] Return sender address & set gas fee

### PR DESCRIPTION
**Pull request by ian-shim**
This PR adds a `SenderAddress` method to `Wallet` interface, which returns the address of the sender.
It also set a `HIGH` gas fee (configured by Fireblocks) to increase the chance of transactions being confirmed sooner.

### CODE REVIEW

File: chainio/clients/fireblocks/contract_call.go [hunk 1]
```diff
@@ -16,6 +17,10 @@ const (
 	Burn         TransactionOperation = "BURN"
 	TypedMessage TransactionOperation = "TYPED_MESSAGE"
 	Raw          TransactionOperation = "RAW"
+
+	FeeLevelHigh   FeeLevel = "HIGH"
+	FeeLevelMedium FeeLevel = "MEDIUM"
+	FeeLevelLow    FeeLevel = "LOW"
```
Comment (samlaf) [inline chainio/clients/fireblocks/contract_call.go:23]: fireblocks doesn't support setting a number for gasPrice?

File: chainio/clients/fireblocks/get_asset_addresses.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+func (f *client) GetAssetAddresses(ctx context.Context, vaultID string, assetID AssetID) ([]AssetAddress, error) {
+	f.logger.Debug("Fireblocks get asset addressees", "vaultID", vaultID, "assetID", assetID)
+	url := fmt.Sprintf("/v1/vault/accounts/%s/%s/addresses", vaultID, assetID)
+	res, err := f.makeRequest(ctx, "GET", url, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error making request: %w", err)
+	}
+	var addresses []AssetAddress
+	err = json.NewDecoder(strings.NewReader(string(res))).Decode(&addresses)
```
Comment (samlaf) [inline chainio/clients/fireblocks/get_asset_addresses.go:29]: any reason you're using json.NewDecoder instead of json.Unmarshal?

File: chainio/clients/wallet/fireblocks_wallet.go [hunk 3]
```diff
@@ -159,6 +159,8 @@ func (t *fireblocksWallet) SendTransaction(ctx context.Context, tx *types.Transa
 		tx.Value().String(),       // amount
 		hexutil.Encode(tx.Data()), // calldata
 		replaceTxByHash,           // replaceTxByHash
+		// TODO: make this configurable
+		fireblocks.FeeLevelHigh, // feeLevel
```
Comment (samlaf) [inline chainio/clients/wallet/fireblocks_wallet.go:163]: why not pass it as argument to SendTransaction?

File: chainio/clients/wallet/wallet.go [hunk 4]
```diff
@@ -16,4 +17,5 @@ type TxID = string
 type Wallet interface {
 	SendTransaction(ctx context.Context, tx *types.Transaction) (TxID, error)
 	GetTransactionReceipt(ctx context.Context, txID TxID) (*types.Receipt, error)
+	SenderAddress(ctx context.Context) (common.Address, error)
```
Comment (samlaf) [inline chainio/clients/wallet/wallet.go:20]: Why is this needed? Can you add comment explaining its usecase. Doesn't seem like its used anywhere in this PR, unless I missed it.

File: chainio/clients/fireblocks/contract_call.go (hunk 1 above)
Comment (ian-shim) [reply chainio/clients/fireblocks/contract_call.go:23]: It does, and we should add them too in the future, but I wanted to just use the fees Fireblocks estimates for us given the fee level

File: chainio/clients/fireblocks/get_asset_addresses.go (hunk 2 above)
Comment (ian-shim) [reply chainio/clients/fireblocks/get_asset_addresses.go:29]: Not exactly sure what the difference is. It was what fireblocks example code was using 😆

File: chainio/clients/wallet/fireblocks_wallet.go (hunk 3 above)
Comment (ian-shim) [reply chainio/clients/wallet/fireblocks_wallet.go:163]: `SendTransaction` should share a common signature for all wallets. `feeLevel` field is specific to Fireblocks wallet.

File: chainio/clients/wallet/wallet.go (hunk 4 above)
Comment (ian-shim) [reply chainio/clients/wallet/wallet.go:20]: I wanted to use this when you pass in the sender address when you create the txMgr

File: chainio/clients/fireblocks/contract_call.go (hunk 1 above)
Comment (samlaf) [reply chainio/clients/fireblocks/contract_call.go:23]: sgtm. Can you just add a TODO comment then to support those, unless its already written somewhere else?

File: chainio/clients/fireblocks/get_asset_addresses.go (hunk 2 above)
Comment (samlaf) [reply chainio/clients/fireblocks/get_asset_addresses.go:29]: haha sounds good, was just curious

File: chainio/clients/wallet/fireblocks_wallet.go (hunk 3 above)
Comment (samlaf) [reply chainio/clients/wallet/fireblocks_wallet.go:163]: hmm I see. so when/how would you set the fee level then? Say if we calculate a gasPrice or whatever? Also what does adding this feeLevel do? Is fireblocks going to replace the gasPrice set in the tx??

File: chainio/clients/wallet/wallet.go (hunk 4 above)
Comment (samlaf) [reply chainio/clients/wallet/wallet.go:20]: yep ok just saw this in avs-sync! Ok that makes sense. Can you just add a comment here explaining that this is needed to create a txMgr that wraps the wallet, since the wallet hides the signer key/address, whereas the txMgr needs to know the from address (forget why though lol)

File: chainio/clients/wallet/fireblocks_wallet.go (hunk 3 above)
Comment (ian-shim) [reply chainio/clients/wallet/fireblocks_wallet.go:163]: Right now, it sends every transaction with `HIGH` gas. There is a TODO for making it configurable. 
If you look at how `SendTransaction` works in this method, it only uses `To`, `Hash`, `Value`, and `Data` of the input transaction. So Fireblocks isn't aware of the gas specified in the `tx` object unless it's passed in in the API call. Right now, it's ignoring the gas fees set in the `tx` and always sending them with `HIGH` fee.

File: chainio/clients/wallet/wallet.go (hunk 4 above)
Comment (ian-shim) [reply chainio/clients/wallet/wallet.go:21]: done

File: chainio/clients/fireblocks/contract_call.go (hunk 1 above)
Comment (ian-shim) [reply chainio/clients/fireblocks/contract_call.go:23]: done

File: chainio/clients/wallet/fireblocks_wallet.go (hunk 3 above)
Comment (samlaf) [reply chainio/clients/wallet/fireblocks_wallet.go:163]: gotcha, thanks for the explanation.

**Review by samlaf — APPROVED** [review]
